tests/unit/widget/widget_classes.js

```diff
@@ -0,0 +1,137 @@
+	options: {
+		classes: {
+			"test-wrapper": "self-wrapper",
+			"test-self": "self-class self-class-2"
+		}
+	},
+	defaultElement: "<div>",
+	_create: function() {
+		this.span = $( "<span>" )
+			.appendTo( this.element );
+
+		this.element.wrap( "<div>" );
+
+		this.wrapper = this.element.parent();
+
+		this._addClass( "test-self", "test-self-extra" )
+			._addClass( "test-self-2" )
+			._addClass( null, "test-self-extra-null" )
+			._addClass( this.span, null, "test-span-extra-null" )
+			._addClass( this.span, "test-span", "test-span-extra" )
+			._addClass( this.wrapper, "test-wrapper" );
+
+	},
+	toggleClasses: function( bool ) {
+		this._toggleClass( "test-self", "test-self-extra", bool )
+			._toggleClass( "test-self-2", null, bool )
+			._toggleClass( null, "test-self-extra-null", bool )
+			._toggleClass( this.span, null, "test-span-extra-null", bool )
+			._toggleClass( this.span, "test-span", "test-span-extra", bool )
+			._toggleClass( this.wrapper, "test-wrapper", null, bool );
+	},
+	removeClasses: function() {
+		this._removeClass( "test-self", "test-self-extra" )
+			._removeClass( "test-self-2" )
+			._removeClass( null, "test-self-extra-null" )
+			._removeClass( this.span, null, "test-span-extra-null" )
+			._removeClass( this.span, "test-span", "test-span-extra" )
+			._removeClass( this.wrapper, "test-wrapper" );
+	},
+	_destroy: function() {
+		this.span.remove();
+		this.element.unwrap();
+	},
+	widget: function() {
+		return this.wrapper;
+	}
+});
+
+test( ".option() - classes setter", function(){
+	expect( 6 );
+
+	var testWidget = $.ui.classesWidget(),
+		currentWrapperClass = testWidget.option( "classes.test-wrapper" );
+
+	testWidget.option({
+		classes: {
+			"test-span": "self-span-new",
+			"test-wrapper": currentWrapperClass + " self-wrapper-new",
+			"test-self": "self-class-2"
+		}
+	});
+
+	equal( testWidget.element.is( ".test-self.self-class-2" ), true,
+		"Removing a class leaves the structure and other classes in value" );
+	equal( !testWidget.element.is( ".self-class" ), true,
+		"Removing a class from the value removes the class" );
+	testWidget.option( "classes.test-self", "" );
+	equal( testWidget.element.is( ".test-self" ), true,
+		"Setting to empty value leaves structure class" );
+	equal( !testWidget.element.is( ".self-class-2" ), true,
+		"Setting empty value removes previous value classes" );
+	equal( testWidget.span.is( ".test-span.self-span-new" ), true,
+		"Adding a class to an empty value works as expected" );
+	equal( testWidget.wrapper.is( ".test-wrapper.self-wrapper-new" ), true,
+		"Appending a class to the current value works as expected" );
+});
+
+test( ".destroy() - class removal", function(){
+	expect( 1 );
+
+	domEqual( "#widget", function(){
+		$( "#widget" ).classesWidget()
+			.classesWidget( "toggleClasses", false )
+			.classesWidget( "toggleClasses", true )
+			.classesWidget( "removeClasses" )
+			.classesWidget( "toggleClasses", true )
+			.classesWidget( "destroy" );
```

Comment:
What would that do? the test widget should already have the classes this would add? The idea here is that they should be removed and re-added to make sure that after destroying EVERYTHING is still removed.